substrate/frame/staking-async/runtimes/papi-tests/src/utils.ts

```diff
@@ -0,0 +1,123 @@
+import { parachain, rc } from "@polkadot-api/descriptors";
+import {
+	Binary,
+	createClient,
+	type PolkadotClient,
+	type PolkadotSigner,
+	type TypedApi,
+} from "polkadot-api";
+import { withPolkadotSdkCompat } from "polkadot-api/polkadot-sdk-compat";
+import { getWsProvider } from "polkadot-api/ws-provider/web";
+import { createLogger, format, log, transports } from "winston";
+import { sr25519CreateDerive } from "@polkadot-labs/hdkd";
+import { DEV_PHRASE, entropyToMiniSecret, mnemonicToEntropy } from "@polkadot-labs/hdkd-helpers";
+import { getPolkadotSigner } from "polkadot-api/signer";
+
+export const GlobalTimeout = 30 * 60 * 1000;
+export const aliceStash = "5GNJqTPyNqANBkUVMN1LPPrxXnFouWXoe2wNSmmEoLctxiZY";
+
+
+export const logger = createLogger({
+	level: process.env.LOG_LEVEL || "verbose",
+	format: format.combine(format.timestamp(), format.cli()),
+	defaultMeta: { service: "staking-papi-tests" },
+	transports: [new transports.Console()],
+});
+
+const miniSecret = entropyToMiniSecret(mnemonicToEntropy(DEV_PHRASE));
+const derive = sr25519CreateDerive(miniSecret);
+const aliceKeyPair = derive("//Alice");
+export const alice = getPolkadotSigner(aliceKeyPair.publicKey, "Sr25519", aliceKeyPair.sign);
+
+export type ApiDeclerations = {
+	rcClient: PolkadotClient;
+	paraClient: PolkadotClient;
+	rcApi: TypedApi<typeof rc>;
+	paraApi: TypedApi<typeof parachain>;
+};
+
+export async function nullifySigned(
+	paraApi: TypedApi<typeof parachain>,
+	signer: PolkadotSigner = alice
+): Promise<boolean> {
+	// signed and signed validation phase to 0
+	const call = paraApi.tx.System.set_storage({
+		items: [
+			// SignedPhase key
+			[
+				Binary.fromBytes(
+					Uint8Array.from([
+						99, 88, 172, 210, 3, 94, 196, 187, 134, 63, 169, 129, 224, 193, 119, 185,
```

Comment:
Could you explain how to derive these? In the `README` there's a link to `https://paritytech.github.io/polkadot-sdk/master/src/frame_support/lib.rs.html#357`, but it's not too clear (to me, at least).

Reply:
Given 

```
parameter_types! {
  pub storage Foo: u32 = 10;
}
```

The key is `xxhas(:Foo:)`. Do you think it is self explanatory enough, or you prefer me to add more info about it here?